### PR TITLE
Document some new tasks in MSBuild 15.8

### PR DESCRIPTION
**Pull request by jeffkl**
* `DownloadFile` task (https://github.com/Microsoft/msbuild/pull/3297)
* `ZipDirectory` task (https://github.com/Microsoft/msbuild/pull/3291)
* `Unzip` task (https://github.com/Microsoft/msbuild/pull/3291)
* `RoslynCodeTaskFactory` (https://github.com/Microsoft/msbuild/pull/3175)

### CODE REVIEW

**Comment by PRMerger12** [comment]
@jeffkl : Thanks for your contribution! The author, @Mikejo5000, has been notified to review your proposed change.

File: docs/msbuild/msbuild-inline-tasks.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ MSBuild tasks are typically created by compiling a class that implements the <xr
   
  Starting in .NET Framework version 4, you can create tasks inline in the project file. You do not have to create a separate assembly to host the task. This makes it easier to keep track of source code and easier to deploy the task. The source code is integrated into the script.  
   
+
+ In MSBuild 15.8, the [RoslnCodeTaskFactory](../msbuild/msbuild-roslyncodetaskfactory.md) was added which can create .NET Standard cross-platform inline tasks.  If you need to use inline tasks on .NET Core, you must use the RoslynCodeTaskFactory.
```
Comment (cdmihai) [inline docs/msbuild/msbuild-inline-tasks.md]: Is it worth saying this is now the preferred way of doing inline tasks? Is it the new preferred way? :)

File: docs/msbuild/msbuild-roslyncodetaskfactory.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+-   `ParameterType` is an optional attribute that is `System.String` by default. It may be set to any fully qualified type that is either an item or a value that can be converted to and from a string by using System.Convert.ChangeType. (In other words, any type that can be passed to and from an external task.)  
+  
+-   `Output` is an optional attribute that is `false` by default. If `true`, then the parameter must be given a value before returning from the Execute method.  
+  
+ For example,  
+  
+```xml  
+<ParameterGroup>  
+    <Expression Required="true" />  
+      <Files ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
```
Comment (cdmihai) [inline docs/msbuild/msbuild-roslyncodetaskfactory.md]: Bad indentation.

File: docs/msbuild/downloadfile-task.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+---
+title: "DownloadFile Task | Microsoft Docs"
+ms.custom: ""
+ms.date: "11/04/2016"
+ms.reviewer: ""
+ms.suite: ""
+ms.technology: msbuild
+ms.tgt_pltfrm: ""
+ms.topic: "article"
```
[the remark this answers is not in the document]
Comment (Mikejo5000) [reply docs/msbuild/downloadfile-task.md]: @jeffkl Reading closer. Some are reference docs so ms.topic would be "reference"

**Comment by Mikejo5000** [comment]
@jeffkl Jeff, for the new topics we will need them in toc.yml before they can get through the approval process. Can you add them? Looks like the ref topics go here: https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-task-reference

File: docs/msbuild/msbuild-inline-tasks.md (hunk 1 above)
Comment (jeffkl) [reply docs/msbuild/msbuild-inline-tasks.md]: Its required for inline tasks on .NET Core but existing builds can just stick with `CodeTaskFactory`.

File: docs/msbuild/downloadfile-task.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeffkl) [reply docs/msbuild/downloadfile-task.md]: Done!

**Comment by Mikejo5000** [comment]
#sign-off

**Comment by ktoliver** [comment]
@jeffkl @Mikejo5000 

Microsoft authors of technical articles need to add content and make updates through the private repository so the updates can be staged and validated by the current validation rules. Read about [authors working in the public repo](https://review.docs.microsoft.com/en-us/help/contribute/contribute-authors-in-public-repo?branch=master) for more information.

Would you submit your updates through the private repo and close this pull request?

Thank you.

**Comment by jeffkl** [comment]
@ktoliver I get a 404 when visiting the private repo and I'm assuming its a permissions issue.  The doc you linked to has no mention of how to request access and I can't figure it out.  How do I request access?

**Comment by Mikejo5000** [comment]
@jeffkl To submit to the private repo, you will need a fork first. The only reason I can think of that you would get a 404 when trying to open it is if you are not on corpnet. @ktoliver Please advise. I haven't seen this requirement before.

**Pull request closed without merging**